### PR TITLE
Create bucket inside Slurm cluster wrapper to control default name

### PR DESCRIPTION
**Pull request by mr0re1**
Create bucket inside Slurm cluster wrapper to set a unique name for the bucket.

### CODE REVIEW

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf [hunk 1]
```diff
@@ -40,6 +40,31 @@ locals {
...
 }
 
+locals {
+  synt_suffix       = substr(md5("${var.project_id}${var.deployment_name}"), 0, 5)
+  synth_bucket_name = "${local.slurm_cluster_name}${local.synt_suffix}"
+}
+
+module "bucket" {
+  source  = "terraform-google-modules/cloud-storage/google"
+  version = "~> 3.0"
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:50]: nit: elsewhere in toolkit we reference 3.4 and 4.0 as a min. Maybe pick similar for consistency.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf [hunk 2]
```diff
@@ -40,6 +40,31 @@ locals {
...
+  names      = [local.synth_bucket_name]
+  prefix     = "slurm"
+  project_id = var.project_id
+
+  force_destroy = {
+    (local.synth_bucket_name) = true
+  }
+
+  labels = {
+    slurm_cluster_name = local.slurm_cluster_name
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:64]: nit: Ideally we would add resource tracking labels here. I notice that controller does not take labels. I think it should.

I am fine to add in separate PR, but please follow up.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf [hunk 3]
```diff
@@ -40,6 +40,31 @@ locals {
 
 }
 
+locals {
+  synt_suffix       = substr(md5("${var.project_id}${var.deployment_name}"), 0, 5)
+  synth_bucket_name = "${local.slurm_cluster_name}${local.synt_suffix}"
+}
+
+module "bucket" {
+  source  = "terraform-google-modules/cloud-storage/google"
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:49]: nit: my personal preference is to use the resource directly (see startup-script as example) instead of cft. I don't really see cft adding value here and it is just one more layer. But we do both in the toolkit so I am fine to leave it.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf [hunk 4]
```diff
@@ -111,17 +136,16 @@ module "slurm_cluster" {
...
+  bucket_name   = var.create_bucket ? module.bucket[0].name : var.bucket_name
   bucket_dir    = var.bucket_dir
 
   controller_instance_config = local.controller_instance_config
 
   enable_login = var.enable_login
   login_nodes  = var.login_nodes
 
-  nodeset     = var.nodeset
-  nodeset_tpu = var.nodeset_tpu
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:124]: This seems unrelated. Why is it deleted?

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:1]: Add rational for this in description: Unique naming of bucket.

**Review by nick-stroud — APPROVED** [review]
Approving to not hold up submission. Please respond to/disposition comments.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf (hunk 2 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:64]: tracking as a separate issue.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf (hunk 4 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:124]: hmm, strange rebase, reverted, thank you for spotting.

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf (hunk 1 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:50]: I'm sticking to the same version used in SchedMD module, would keep it this way until we rethink this whole situation (hopefully soon).

File: community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf (hunk 3 above)
Comment (mr0re1) [reply community/modules/scheduler/schedmd-slurm-gcp-v6-controller/main.tf:49]: I agree, though this is quick fix without changes to the interfaces, I'm open to change it near future.

**Review by nick-stroud — APPROVED** [review]

**Review by nick-stroud — APPROVED** [review]
